src/beamlime/applications/handlers.py

```diff
@@ -27,23 +34,50 @@ class WorkflowResultUpdate:
     content: WorkflowResult
 
 
+def nth_or_maxtime(n, maxtime):
+    count = 0
+    last = time.time()
+
+    def run():
+        nonlocal count, last
+        count += 1
+        if count >= n or time.time() - last >= maxtime:
+            count = 0
+            last = time.time()
+            return True
+
+    return run
+
+
 class DataAssembler(HandlerInterface):
     """Receives data and assembles it into a single data structure."""
 
-    def __init__(self) -> None:
-        from ._nexus_helpers import NexusContainer
-
-        self.nexus_container: NexusContainer
+    def __init__(self, merge_every_nth=float('inf'), max_seconds_between_messages=5):
+        self._store = {}
```

Comment:
Can you add more explanation what the `_store` is?
Or make it as a typed dict...?

Reply:
It's a `Mapping[Tuple[str, ...], Mapping]`, a mapping from paths to nexus-json group objects.

It stores the nexus group objects created by the merged messages, and the paths where they should be inserted into the structure.

After a `DataReady` messages is emitted the store is reset so that the next `DataReady` message doesn't contain data from earlier messages.

Reply:
I'll add that type annotation to the code